episodes/13-synchronization.md

```diff
@@ -81,67 +78,68 @@ thread 2: 7
 thread 2: 8
 thread 2: 9
 thread 2: 10
-~~~
-{: .output}
+```
 
-> ## Discussion
->
-> What would happen if we write instead 
-> ~~~
-> begin
-> {
->   sync
->   {
->     var c=0;
->     while c<10
->     {
->       c+=1;
->       writeln('thread 1: ',x+c);
->     }
->   }
-> }
-> writeln("The first task is done...");
-> ~~~
-> {: .source}
-{: .discussion}
-
-> ## Exercise 3
->
-> Use `begin` and `sync` statements to reproduce the functionality of `cobegin`
-> in `cobegin_example.chpl`.
->
-> > ## Solution
-> > ~~~
-> > var x=0;
-> > writeln("This is the main thread, my value of x is ",x);
-> >
-> > sync
-> > {
-> >     begin
-> >     {
-> >        var x=5;
-> >        writeln("this is task 1, my value of x is ",x);
-> >     }
-> >     begin writeln("this is task 2, my value of x is ",x);
-> >  }
-> >
-> > writeln("this message won't appear until all tasks are done...");
-> > ~~~
-> > {: .source}
-> {: .solution}
-{: .challenge}
-
-A more elaborated and powerful use of `sync` is as a type qualifier for
-variables. When a variable is declared as _sync_, a state that can be
-**_full_** or **_empty_** is associated to it.
-
-To assign a new value to a _sync_ variable, its state must be _empty_ (after
-the assignment operation is completed, the state will be set as _full_). On the
-contrary, to read a value from a _sync_ variable, its state must be _full_
-(after the read operation is completed, the state will be set as _empty_
-again).
-
-~~~
+::::::::::::::::::::::::::::::::::::::: discussion
+
+## Discussion
+
+What would happen if we write instead
+
+```chpl
+begin
+{
+  sync
+  {
+    var c=0;
+    while c<10
+    {
+      c+=1;
+      writeln('thread 1: ',x+c);
+    }
+  }
+}
+writeln("The first task is done...");
+```
+
+:::::::::::::::::::::::::::::::::::::::::::::::::::
+
+::::::::::::::::::::::::::::::::::::: challenge
+
+## Challenge 3: Can you do it?
```

Comment:
Is there a good reason to use `sync` instead of `cobegin`? Pedagogical purpose of this Challenge is unclear.

Reply:
*sync* provides more manual control. I see your point, but I don't see any problem with leaving this challenge in the lesson -- until we have a better one with *sync*. If you can suggest one, that would be awesome, or we could open an issue on this.